kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java

```diff
@@ -611,4 +650,21 @@ public static KubernetesClientException requestException(Request request, Except
   public Config getConfig() {
     return config;
   }
+
+  private MediaType getMediaTypeFromPatchContextOrDefault(PatchContext patchContext) {
+    if (patchContext != null && patchContext.getPatchType() != null) {
+      return getMediaTypeFromPatchType(patchContext.getPatchType());
+    }
+    return STRATEGIC_MERGE_JSON_PATCH;
+  }
+
+  private MediaType getMediaTypeFromPatchType(PatchType patchType) {
```

Comment:
You can improve the PatchType enum so that the MediaType is defined there and remove this:

`patchType.getMediaType()`